src/test/linters/lint-files.unit.test.ts

```diff
@@ -0,0 +1,367 @@
+        origRCFile: basename ? path.join(dirname, basename) : ''
+    };
+}
+
+class TestFixture extends BaseTestFixture {
+    constructor(
+        printLogs = false
+    ) {
+        const serviceContainer = TypeMoq.Mock.ofType<IServiceContainer>(undefined, TypeMoq.MockBehavior.Strict);
+        const configService = TypeMoq.Mock.ofType<IConfigurationService>(undefined, TypeMoq.MockBehavior.Strict);
+
+        const platformService = new PlatformService();
+        const filesystem = new FileSystem(platformService);
+
+        super(
+            platformService,
+            filesystem,
+            TestFixture.newPythonToolExecService(
+                serviceContainer.object
+            ),
+            TestFixture.newPythonExecFactory(
+                serviceContainer,
+                configService.object,
+                platformService
+            ),
+            configService,
+            serviceContainer,
+            workspaceDir,
+            printLogs
+        );
+
+        this.pythonSettings.setup(s => s.pythonPath)
+            .returns(() => 'python2');
+    }
+
+    private static newPythonToolExecService(
+        serviceContainer: IServiceContainer
+    ): IPythonToolExecutionService {
+        // We do not worry about the IProcessServiceFactory possibly
+        // needed by PythonToolExecutionService.
+        return new PythonToolExecutionService(
+            serviceContainer
+        );
+    }
+
+    private static newPythonExecFactory(
+        serviceContainer: TypeMoq.IMock<IServiceContainer>,
+        configService: IConfigurationService,
+        platformService: IPlatformService
+    ): IPythonExecutionFactory {
+        const envVarsService = TypeMoq.Mock.ofType<IEnvironmentVariablesProvider>(undefined, TypeMoq.MockBehavior.Strict);
+        envVarsService.setup(e => e.getEnvironmentVariables(TypeMoq.It.isAny()))
+            .returns(() => Promise.resolve({}));
+        serviceContainer.setup(c => c.get(TypeMoq.It.isValue(IEnvironmentVariablesProvider), TypeMoq.It.isAny()))
+            .returns(() => envVarsService.object);
+
+        const envActivationService = TypeMoq.Mock.ofType<IEnvironmentActivationService>(undefined, TypeMoq.MockBehavior.Strict);
+
+        const decoder = new BufferDecoder();
+        serviceContainer.setup(c => c.get(TypeMoq.It.isValue(IBufferDecoder), TypeMoq.It.isAny()))
+            .returns(() => decoder);
+
+        const procServiceFactory = new ProcessServiceFactory(serviceContainer.object);
+
+        return new PythonExecutionFactory(
+            serviceContainer.object,
+            envActivationService.object,
+            procServiceFactory,
+            configService,
+            decoder
+        );
+    }
+
+    public makeDocument(filename: string): TextDocument {
+        const doc = newMockDocument(filename);
+        return doc.object;
+    }
+}
+
+// tslint:disable-next-line:max-func-body-length
+suite('Linting - functional (with real files)', () => {
```

Comment:
If this is a suite that tests real files on the system, then this is not a unit test and should be in a system test file, not a unit test one.

I feel that it is important that we strictly separate the two kinds of tests.
```suggestion
suite('Linting Functional Tests', () => {
```

Reply:
fixed